pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -71,42 +73,75 @@ func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetwork
 	needReboot = false
 	err = nil
 	attributesToChange = map[string]mlnxNic{}
+	mellanoxNicsStatus = map[string]map[string]*sriovnetworkv1.InterfaceExt{}
+	mellanoxNicsSpec = map[string]*sriovnetworkv1.Interface{}
```

Comment:
You have package scoped variables with the same name. These shall be removed.

Reply:
> You have package scoped variables with the same name. These shall be removed.

The `mellanoxNicsStatus` needs to be initialized once so moved line 76 to the `init` function (it will case panic if it wasn't initialized `panic: assignment to entry in nil map`)

The `mellanoxNicsSpec` needs to be updated with the new specs every time `OnNodeStateChange` is called